plugins/crate-copy-azure/src/test/java/io/crate/copy/azure/OperatorHelperTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.copy.azure;
+
+import static io.crate.copy.azure.AzureBlobStorageSettings.KEY_SETTING;
+import static io.crate.copy.azure.AzureBlobStorageSettings.PROTOCOL_SETTING;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.net.URI;
+
+import org.elasticsearch.common.settings.Settings;
+import org.junit.Test;
+
+public class OperatorHelperTest {
+
+    private static final Settings.Builder COMMON_SETTING_BUILDER = Settings.builder()
+        .put(KEY_SETTING.getKey(), "dummy"); // Dummy value to pass auth validation.
+
+    @Test
+    public void test_unknown_protocol_is_rejected() throws Exception {
+        Settings settings = COMMON_SETTING_BUILDER.put(PROTOCOL_SETTING.getKey(), "pg").build();
+
+        URI uri = URI.create("az://myaccount.blob.core.windows.net/my-container/dir");
+        assertThatThrownBy(() -> OperatorHelper.config(AzureURI.of(uri), settings, true))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Invalid protocol `pg`. Expected HTTP or HTTPS");
+    }
```

Comment:
This isn't testing the `OperatorHelper` but just the `PROTOCOL_SETTING` inside `AzureBlobStorageSettings`. It should be moved.

Reply:
Validation is triggered only when we access the setting via setting.get(settings), validate doesn't access it. 

But yeah, it's better to be in settings tests -> used `PROTOCOL_SETTING.get(settings)` directly.